sdk/tests/deqp/functional/gles3/es3fShaderStateQueryTests.js

```diff
@@ -0,0 +1,2241 @@
+        ['isampler2DArray', '', 'highp', '', 'float(textureSize(uniformValue,0).r)', gl.INT_SAMPLER_2D_ARRAY, 1, false],
+        ['usampler2D', '', 'highp', '', 'float(textureSize(uniformValue,0).r)', gl.UNSIGNED_INT_SAMPLER_2D, 1, false],
+        ['usampler3D', '', 'highp', '', 'float(textureSize(uniformValue,0).r)', gl.UNSIGNED_INT_SAMPLER_3D, 1, false],
+        ['usamplerCube', '', 'highp', '', 'float(textureSize(uniformValue,0).r)', gl.UNSIGNED_INT_SAMPLER_CUBE, 1, false],
+        ['usampler2DArray', '', 'highp', '', 'float(textureSize(uniformValue,0).r)', gl.UNSIGNED_INT_SAMPLER_2D_ARRAY, 1, false]
+    ];
+
+    var vertSource =
+        '#version 300 es\n' +
+        'void main (void)\n' +
+        '{\n' +
+        ' gl_Position = vec4(0.0);\n' +
+        '}\n';
+
+    var shaderVert = gl.createShader(gl.VERTEX_SHADER);
+    var shaderFrag = gl.createShader(gl.FRAGMENT_SHADER);
+    var program = gl.createProgram();
+
+    gl.attachShader(program, shaderVert);
+    gl.attachShader(program, shaderFrag);
+
+    gl.shaderSource(shaderVert, vertSource);
+    gl.compileShader(shaderVert);
+
+    for (var ndx = 0; ndx < uniformTypes.length; ++ndx) {
+        var declaration = uniformTypes[ndx][0];
+        var postDeclaration = uniformTypes[ndx][1];
+        var precision = uniformTypes[ndx][2];
+        var layout = uniformTypes[ndx][3];
+        var getter = uniformTypes[ndx][4];
+        var type = uniformTypes[ndx][5];
+        var size = uniformTypes[ndx][6];
+        var isRowMajor = uniformTypes[ndx][7];
+        bufferedLogToConsole('Verify type of ' + declaration + ' variable' + postDeclaration);
+
+        // gen fragment shader
+
+        var frag = '';
+        frag += '#version 300 es\n';
+        frag += layout + 'uniform ' + precision + ' ' + declaration + ' uniformValue' + postDeclaration + ';\n';
+        frag += 'layout(location = 0) out mediump vec4 fragColor;\n';
+        frag += 'void main (void)\n';
+        frag += '{\n';
+        frag += ' fragColor = vec4(' + getter + ');\n';
+        frag += '}\n';
+
+        gl.shaderSource(shaderFrag, frag);
+
+        // compile & link
+
+        gl.compileShader(shaderFrag);
+        gl.linkProgram(program);
+
+        // test
+        if (this.check(glsStateQuery.verifyProgram(program, gl.LINK_STATUS, true), 'Program link fail' + gl.getProgramInfoLog(program))) {
+            var indices = gl.getUniformIndices(program, ['uniformValue']);
+            var info_type = gl.getActiveUniforms(program, indices, gl.UNIFORM_TYPE)[0];
+            var info_size = gl.getActiveUniforms(program, indices, gl.UNIFORM_SIZE)[0];
+            var info_is_row_major = gl.getActiveUniforms(program, indices, gl.UNIFORM_IS_ROW_MAJOR)[0];
+            this.check(glsStateQuery.compare(info_size, size));
+            this.check(glsStateQuery.compare(info_type, type));
+            this.check(glsStateQuery.compare(info_is_row_major, isRowMajor));
```

Comment:
I have a doubt here. According to the spec gl.UNIFORM_IS_ROW_MAJOR is a GLint.
IMO the spec should better be changed and gl.UNIFORM_IS_ROW_MAJOR should be boolean. The test in its current shape expects it to be boolean.